components/tools/measure.js

```diff
@@ -3,377 +3,170 @@
-                    nearest = data;
-                }
+                container.select('rect')
+                    .attr('x', measure.xScale.value.range()[0])
+                    .attr('y', measure.yScale.value.range()[1])
+                    .attr('width', measure.xScale.value.range()[1])
+                    .attr('height', measure.yScale.value.range()[0]);
+
+                var g = fc.utilities.simpleDataJoin(container, 'measure', data);
+
+                var enter = g.enter();
+                enter.append('line')
+                    .attr('class', 'tangent');
+                enter.append('line')
+                    .attr('class', 'horizontal');
+                enter.append('line')
+                    .attr('class', 'vertical');
+                enter.append('text')
+                    .attr('class', 'horizontal');
+                enter.append('text')
+                    .attr('class', 'vertical');
+
+                g.select('line.tangent')
+                    .attr('x1', function(d) { return d.source.x; })
+                    .attr('y1', function(d) { return d.source.y; })
+                    .attr('x2', function(d) { return d.target.x; })
+                    .attr('y2', function(d) { return d.target.y; });
+
+                g.select('line.horizontal')
+                    .attr('x1', function(d) { return d.source.x; })
+                    .attr('y1', function(d) { return d.source.y; })
+                    .attr('x2', function(d) { return d.target.x; })
+                    .attr('y2', function(d) { return d.source.y; })
+                    .style('visibility', function(d) { return d.state !== 'MEASURED' ? 'hidden' : 'visible'; });
+
+                g.select('line.vertical')
+                    .attr('x1', function(d) { return d.target.x; })
+                    .attr('y1', function(d) { return d.target.y; })
+                    .attr('x2', function(d) { return d.target.x; })
+                    .attr('y2', function(d) { return d.source.y; })
+                    .style('visibility', function(d) { return d.state !== 'MEASURED' ? 'hidden' : 'visible'; });
+
+                var paddingValue = measure.padding.value.apply(this, arguments);
+
+                g.select('text.horizontal')
+                    .attr('x', function(d) { return d.source.x + (d.target.x - d.source.x) / 2; })
+                    .attr('y', function(d) { return d.source.y - paddingValue; })
+                    .style('visibility', function(d) { return d.state !== 'MEASURED' ? 'hidden' : 'visible'; })
+                    .text(measure.xLabel.value);
+
+                g.select('text.vertical')
+                    .attr('x', function(d) { return d.target.x + paddingValue; })
+                    .attr('y', function(d) { return d.source.y + (d.target.y - d.source.y) / 2; })
+                    .style('visibility', function(d) { return d.state !== 'MEASURED' ? 'hidden' : 'visible'; })
+                    .text(measure.yLabel.value);
+
```

Comment:
As a general note I think we'll eventually need to revisit the use of data in the tools. Whilst it is nice, in that it provides the decorate pattern, now that we know a little more about selections I think we can easily fake that behaviour and simplify our code.